src/vivarium/framework/engine.py

```diff
@@ -46,13 +46,39 @@
 
 
 class SimulationContext:
+
+    _created_simulation_contexts: Set[str] = set()
+
+    @staticmethod
+    def _get_context_name(sim_name: Union[str, None]) -> str:
+        # Not threadsafe, but I don't see why we'd need to be.
+        if sim_name:
+            if sim_name in SimulationContext._created_simulation_contexts:
+                msg = (
+                    "Attempting to create two SimulationContexts "
+                    f"with the same name {sim_name}"
+                )
+                raise VivariumError(msg)
+
+        if sim_name is None:
+            sim_number = len(SimulationContext._created_simulation_contexts) + 1
+            sim_name = f"simulation_{sim_number}"
```

Comment:
What if there was a simulation manually named this? You should move this assignment above the error check.

Reply:
good catch